blog/views.py

```diff
@@ -54,3 +54,49 @@ def blog_delete(request, pk):
         return redirect('blog:blog_list')
 
     return render(request, 'blog/delete.html', {'post': post})
+
+@login_required
+def post_detail(request, slug, year, month, date, post):
+    template_name = 'post_detail.html'
+    post = get_object_or_404(Post, slug=post,
+                             status='published',
+                             publish_year=year,
+                             publish_month=month,
+                             publish_day=day)
+
+    # List of active comments for this post
+    comments = post.comments.filter(active=True)
+
+    new_comment = None
+
+    if request.method == 'POST':
+        # A comment was posted
+        comment_form = CommentForm(date=request.POST)
+        if comment_form_is_valid():
+            #Create Comment object but doesn't save database yet
+            new_comment = comment_form.save(comment=False)
+            # Assign the current post to the comment
+            new_comment.post = post
+            # Save comment to the database
+            new_comment.save()
+        else:
+            comment_form = CommentForm()
```

Comment:
I think you can drop this line, because it would give you a blank form and ideally you want to pass back the filled in values back to the template.